src/Microsoft.OData.ModelBuilder/Vocabularies/Capabilities/DeleteRestrictionsType.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.OData.ModelBuilder.Vocabularies.Capabilities
+{
+    /// <summary>
+    /// Restrictions on delete operations
+    /// </summary>
+    public class DeleteRestrictionsType
```

Comment:
>DeleteRestrictionsType [](start = 17, length = 22)

for brainstorm:

we hard cording the restrictions type, it works. However, everytime, we change the type definition, we have to change it.
So, can we codegen the CLR types from the Edm complex type definition, so, we don't need to hard coding the types.

Reply:
I had the same idea although I nixed it for now as currently only focusing on the subset of the vocabulary that deal with CRUD (and operation) restrictions and for those, currently only focusing on the subset pf properties that are dealing directly with the restrictions ie whether the action is allowed plus the permissions definition as is required by OData WebApi...
We could look at code generation when dealing with the whole vocabulary(ies) in #5